newIDE/app/src/Profile/Subscription/PlanCard.js

```diff
@@ -14,59 +18,197 @@ import { useResponsiveWindowWidth } from '../../UI/Reponsive/ResponsiveWindowMea
+        return (
+          <Text key="year" noMargin color="secondary">
+            <Trans>
+              {price.amountInCents / 100}
+              {price.currency === 'EUR' ? '€' : '$'}/seat every{' '}
+              {price.periodCount} years
+            </Trans>
+          </Text>
+        );
+      } else {
+        return (
+          <Text key="year" noMargin color="secondary">
+            <Trans>
+              {price.amountInCents / 100}
+              {price.currency === 'EUR' ? '€' : '$'} every {price.periodCount}{' '}
+              years
+            </Trans>
+          </Text>
+        );
+      }
+    }
+  }
+};
+
+/**
+ * If price is given, it will be displayed. Otherwise, the prices stored in the
+ * subscription plan will be displayed.
+ */
+const getPlanPrices = ({
+  prices,
+  price,
   hidePrice,
 }: {
-  plan: PlanDetails,
+  prices: ?(SubscriptionPlanPrice[]),
+  price: ?SubscriptionPlanPrice,
   hidePrice?: boolean,
 }): React.Node => {
-  if (hidePrice || plan.monthlyPriceInEuros === null) return null;
-  if (plan.monthlyPriceInEuros === 0)
-    return (
-      <Text noMargin color="secondary">
-        <Trans>Free</Trans>
-      </Text>
-    );
-  const prices = [];
-  prices.push(
-    plan.isPerUser ? (
-      <Text key="month" noMargin color="secondary">
-        <Trans>{plan.monthlyPriceInEuros}€/seat/month</Trans>
-      </Text>
-    ) : (
-      <Text key="month" noMargin color="secondary">
-        <Trans>{plan.monthlyPriceInEuros}€/month</Trans>
-      </Text>
-    )
-  );
-  if (plan.yearlyPriceInEuros) {
-    prices.push(
-      plan.isPerUser ? (
-        <Text key="year" noMargin color="secondary">
-          <Trans>or {plan.yearlyPriceInEuros}€/seat/year</Trans>
-        </Text>
-      ) : (
-        <Text key="year" noMargin color="secondary">
-          <Trans>or {plan.yearlyPriceInEuros}€/year</Trans>
-        </Text>
-      )
-    );
+  if (hidePrice) return null;
+  if (price) {
+    return getPlanPrice(price);
+  }
+
+  if (prices && prices.length > 0) {
+    const displayedPrices = prices
+      .sort((priceA, priceB) => {
+        // TODO: Decide if the front end should sort the prices or if it should respect
+        // the order sent by the back end.
+        if (priceA.period === 'week' && priceB.period === 'month') return 1;
+        if (priceA.period === 'week' && priceB.period === 'year') return 1;
+        if (priceA.period === 'month' && priceB.period === 'year') return 1;
+        if (priceA.period === priceB.period) return 0;
+        return -1;
+      })
+      .map(getPlanPrice);
+
+    return displayedPrices;
   }
-  return prices;
+
+  return (
+    <Text noMargin color="secondary">
+      <Trans>Free</Trans>
+    </Text>
+  );
 };
 
 const PLAN_LOGO_SIZE = 25;
 const PLAN_LOGO_PADDING = 20;
 
-const getPlanIcon = (plan: PlanDetails): React.Node => {
-  switch (plan.planId) {
+const getPlanIcon = (
+  subscriptionPlan: SubscriptionPlan | SubscriptionPlanWithPrices
```

Comment:
following the other comment in the typings in Usage.js
It's confusing to handle 2 types here. There should probably only be 1?